api/v1/_submissions/PKPBackendSubmissionsController.php

```diff
@@ -87,6 +87,15 @@ public function getGroupRoutes(): void
                 ),
             ]);
 
+        Route::delete('', $this->bulkDeleteIncompleteSubmissions(...))
+            ->name('_submission.incomplete.delete')
+            ->middleware([
+                self::roleAuthorizer([
+                    Role::ROLE_ID_SITE_ADMIN,
+                    Role::ROLE_ID_MANAGER,
```

Comment:
Without the `Role:: ROLE_ID_MANAGER` entry for the middleware, requests with only a `Role::ROLE_ID_SITE_ADMIN` fails with a 401 error -  {"error":"You are not authorized to access the requested resource."}.


There is also a previous delete route, to delete a declined submission, that fails with the same error if I update the middleware to only use `Role::ROLE_ID_SITE_ADMIN` and attempt to delete a submission as an admin

https://github.com/pkp/pkp-lib/blob/3704af0dc5ce4454c3f0dbddde2a9734cb1351c5/api/v1/_submissions/PKPBackendSubmissionsController.php#L90-L99

Reply:
@taslangraham and @touhidurabir, I think there was probably an issue introduced with https://github.com/pkp/pkp-lib/issues/7698.

Before #7698 was introduced, everywhere that the `ROLE_ID_MANAGER` role was checked, `ROLE_ID_SITE_ADMIN` was checked too -- either was sufficient to grant access. (This was introduced with [#7392](https://github.com/pkp/pkp-lib/issues/7392#issuecomment-1119184605.)) But once [`HasRoles`](https://github.com/pkp/pkp-lib/blob/main/classes/middleware/HasRoles.php) replaced calls to [`PKPBaseController::addRoleAssignment`](https://github.com/pkp/pkp-lib/issues/7698), `ROLE_ID_SITE_ADMIN` is never getting added to the role list because [`$user->getRoles`](https://github.com/pkp/pkp-lib/blob/22af741d448e394dcbaeb8cc652530f5705666e1/classes/middleware/HasRoles.php#L50) will never return it when there is a context. (Enrolments with `ROLE_ID_SITE_ADMIN` will always have a context ID of `null`.)

@touhidurabir, could you look into adjusting `HasRoles` to include site admins?

Reply:
@touhidurabir @asmecher I've opened an issue and PR for this.
PR - https://github.com/pkp/pkp-lib/pull/10267

Reply:
The fix from the PR above has been merged